flink-core/src/main/java/org/apache/flink/configuration/RestOptions.java

```diff
@@ -160,6 +160,20 @@
                     .withDescription(
                             "The number of threads for the asynchronous processing of requests.");
 
+    @Documentation.Section(Documentation.Sections.EXPERT_REST)
+    public static final ConfigOption<Integer> JOBMANAGER_FUTURE_THREADS =
+            key("jobmanager.future-executor.numThreads")
+                    .defaultValue(0)
+                    .withDescription(
+                            "The number of jobmanager future-executor threads for shared services.");
+
+    @Documentation.Section(Documentation.Sections.EXPERT_REST)
+    public static final ConfigOption<Integer> JOBMANAGER_IO_THREADS =
+            key("jobmanager.io-executor.numThreads")
+                    .defaultValue(0)
+                    .withDescription(
+                            "The number of jobmanager io-executor threads for shared services.");
```

Comment:
I think we should put these options into the `JobManagerOptions` class.

Reply:
Maybe we could even reuse the `ClusterOptions.CLUSTER_IO_EXECUTOR_POOL_SIZE` for the `ioExecutor`. Then we could add another `jobmaster.future-pool.size` that we also add to `ClusterOptions`.

Reply:
We could then also reuse the `ClusterEntrypointUtils.getPoolSize`.